app/src/main/java/fr/free/nrw/commons/Utils.java

```diff
@@ -303,11 +304,11 @@ public static String fixExtension(String title, String extension) {
         Pattern jpegPattern = Pattern.compile("\\.jpeg$", Pattern.CASE_INSENSITIVE);
 
         // People are used to ".jpg" more than ".jpeg" which the system gives us.
-        if (extension != null && extension.toLowerCase().equals("jpeg")) {
+        if (extension != null && extension.toLowerCase(Locale.US).equals("jpeg")) {
             extension = "jpg";
         }
         title = jpegPattern.matcher(title).replaceFirst(".jpg");
-        if (extension != null && !title.toLowerCase().endsWith("." + extension.toLowerCase())) {
+        if (extension != null && !title.toLowerCase(Locale.US).endsWith("." + extension.toLowerCase(Locale.US))) {
```

Comment:
title can be non-English (US English or anything else) on Wikimedia Commons, so it seems to be better to stick with the default locale.

extension here should be ASCII only, so Locale.US seems to make sense. (But did you also consider using Locale.ROOT?)